app/components/theme/switcher_component.html.erb

```diff
@@ -1,4 +1,40 @@
-<%= link_to themes_path(theme: other_theme.name), class: yass(link: type), data: { turbo_method: :put } do %>
-  <%= inline_svg_tag "icons/#{current_theme.icon}.svg", class: yass(icon: type), aria: true, title: 'theme icon' %>
-  <%= text unless icon_only? %>
-<% end %>
+<div class="relative <% mobile? ? 'dark:hover:bg-gray-700/60 dark:hover:text-gray-200' : '' %>" data-controller="visibility" data-action="visibility:click:outside->visibility#off" data-visibility-visible-value="false">
+  <% if mobile? %>
+    <%= render MobileThemeSwitcherComponent.new(
```

Comment:
We've got this nested within another visibility controller when it doesn't have to, that may cause a few problems.

Can we move up the mobile check to the top level of nesting, the structure of the file should be something like: 

```
<% if mobile %>
  render mobile switcher
<% else %>
  <div class"relative ..." data-controller="visibility">
    ....
  </div>
<% end %>
  ```